operator/v1/types_machineconfiguration.go

```diff
@@ -153,3 +167,181 @@ const (
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicySSHKey is actions to take for any SSHKey change
+type NodeDisruptionPolicySSHKey struct {
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// Validation is a string enum used in a NodeDisruptionPolicyStatus object. They describe whether the controller
+// was able to validate the latest spec
+// +kubebuilder:validation:Enum:="success";"failed"
+type Validation string
+
+const (
+	// success represents a valid cluster node disruption policy
+	Success Validation = "success"
+
+	// failed represents an invalid cluster node disruption policy
+	Failed Validation = "failed"
+)
+
+// NodeDisruptionPolicyType is a string enum used in a NodeDisruptionPolicyConfig object. They describe the type of
+// the object that this policy will be in effect for.
+// +kubebuilder:validation:Enum:="file";"service";"sshKey"
+type NodeDisruptionPolicyType string
+
+const (
+	// File represents a NodeDisruption policy that is in effect for changes to a file.
+	File NodeDisruptionPolicyType = "file"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a service.
+	Service NodeDisruptionPolicyType = "service"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a sshKey
+	SSHKey NodeDisruptionPolicyType = "sshKey"
+)
+
+// TODO maybe do this in the MCC validation and skip here?
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'reload' ? has(self.reload) : !has(self.reload)",message="Reload is required when type is reload, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'restart' ? has(self.restart) : !has(self.restart)",message="Restart is required when type is restart, and forbidden otherwise"
+// +union
+type NodeDisruptionPolicyAction struct {
+	// type represents the commands that will be carried out if this NodeDisruptionPolicyActionType is executed
+	// Valid value(s): reboot, drain, reload, restart, daemon-reload, none, special
+	// reload/restart requires a corresponding service target specified in the reload/restart field. All other fields cannot have a these fields
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NodeDisruptionPolicyActionType `json:"type"`
+	// reload specifies the service to reload, only valid if type is reload
+	// +optional
+	Reload *ReloadService `json:"reload,omitempty"`
+	// restart specifies the service to restart, only valid if type is restart
+	// +optional
+	Restart *RestartService `json:"restart,omitempty"`
+}
+
+// ReloadService allows the user to specify the services to be reloaded
+type ReloadService struct {
+	// ServiceName is the full name (e.g. crio.service) of the service to be reloaded
```

Comment:
Should be lower case at the start of the godoc. Similar questions above about service and the godoc here